ui/packages/dashboard_client/src/index.ts

```diff
@@ -16,6 +17,21 @@ function getBasePath(): string {
   return basePath
 }
 
+export async function reqWithErrPrompt<T>(
```

Comment:
I think you can change current [axios interceptors](https://github.com/pingcap-incubator/tidb-dashboard/blob/master/ui/src/utils/client.js#L21) to introduce this logic.

Note that sometimes we want to handle errors manually, for example, in the sign in page we want to perform custom actions when password is incorrect. Considering this, there are two solutions:

Solution 1. Maybe better to change sign in page to return http code 200 or some other special http code in such scenario.

Solution 2. Pass additional custom configuration (like `skipDefaultErrorHandler`) in sign in page indicating that we want to handle error response manually.

Personally I prefer solution 2 if it works.

Reply:
@breeswish mentor, 
- If we put all error handle logic in axios interceptor, it is hard to pass front end i18n error message to popup
- Maybe we can just prompt what response `error.message` contains, let backend return i18n error message depends on request header like `language=en`.

Reply:
@SSebo 

> If we put all error handle logic in axios interceptor, it is hard to pass front end i18n error message to popup for each request.

Maybe you can just use the `i18next.t()` function, which will behave correctly, following the global i18n settings. Actually now in axios interceptors it is already being used to prompt an expired session message.

Reply:
@breeswish  For example `t('search_logs.error.search_log')`, how to pass `search_logs.error.search_log` into request to know which error message to prompt ?

Reply:
Maybe there is no need, since search log request will only fail when there are network errors and auth issues. If a custom message is needed, we need to follow the way of signIn.